lib/repo-url.js

```diff
@@ -0,0 +1,66 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const { URL } = require('url')
+const hostedGitInfo = require('hosted-git-info')
+
+/**
+ * @param packageName A package name as listed in package.json's dependencies list
+ * @param packageJson Optional param to specify a object representation of a package.json file instead of loading from node_modules
+ * @returns A valid url to the root of the package's source or null if a url could not be determined
+ */
+function getRepoUrl(packageName, packageJson = undefined) {
+  if (packageJson === undefined) {
```

Comment:
I prefer `if (!packageJson)` since it also catches `null` and we don't have to worry about `false` or `0`.

Reply:
Ha there's my TypeScript experience coming in. I try to not coerce values of different types and be explicit. But happy to change it if that's the standard here.

Reply:
If it were in Typescript, the type checker would prevent `null` from being passed. But since we're in vanilla JS, we should handle `null` to guard against a runtime error. 

One of these days I'll get around to converting to Typescript...